### PR TITLE
Fix generate diff script

### PR DESCRIPTION
**Pull request by omichel**
Passing the parent branch as an argument to generate_diff.py script is more scalable.

### CODE REVIEW

**Comment by omichel** [comment]
👍

File: .travis.yml [hunk 1]
```diff
@@ -34,7 +34,10 @@ jobs:
         - pip install -r docs/tests/requirements.txt
       script:
         - python -m unittest discover -s docs/tests
-        - if [ $TRAVIS_EVENT_TYPE != cron ]; then python tests/sources/generate_diff.py; fi
+        - |
+          if [ "$TRAVIS_EVENT_TYPE" != "cron" ] && [ "$TRAVIS_EVENT_TYPE" = "pull_request" ]; then
```
Comment (DavidMansolino) [inline .travis.yml]: These two conditions are redundant, the second condition should be sufficient.

**Comment by omichel** [comment]
:+1:

File: tests/sources/generate_diff.py [hunk 2]
```diff
@@ -14,17 +14,18 @@
...
 import sys
 
-branch = os.getenv('TRAVIS_BRANCH')  # branch targeted by the pull request
-if branch is not None and os.getenv('TRAVIS_EVENT_TYPE') == 'pull_request':
-    output = subprocess.check_output(['git', 'diff', '--name-only', branch]).decode('utf-8')
-    f = open(os.path.join(os.getenv('WEBOTS_HOME'), 'tests', 'sources', 'modified_files.txt'), 'w')
-    f.write(output)
-    f.close()
+if len(sys.argv) != 2:  # no parent branch passed as an argument, computing it from script
+    script = os.getenv("WEBOTS_HOME") + os.sep + 'tests' + os.sep + 'sources' + os.sep + 'parent_branch.sh'
```
Comment (fabienrohrer) [inline tests/sources/generate_diff.py]: Just a minor suggestion to be more concise and coherent with the statement 2 lines below:

```suggestion
    script = os.path.join(os.getenv("WEBOTS_HOME"), 'tests', 'sources', 'parent_branch.sh')
```

**Pull request merged**